### PR TITLE
Display details in notifications

### PR DESCRIPTION
**Pull request by untreu2**
### Invitation Notifications
- **DM invitations**: 
  - Title: Welcomer's display name (falls back to "Unknown")
  - Body: "Invited you to chat"
- **Group invitations**:
  - Title: Welcomer's display name (falls back to "Unknown")
  - Body: "Invited you to {groupName}"

### Message Notifications
- **DM messages**:
  - Title: Sender's display name (falls back to "Unknown")
  - Body: Message content (with 📷 emoji prefix if media is present)
- **Group messages**:
  - Title: Group name
  - Body: "{Sender name}: {message content}" (with 📷 emoji prefix if media is present)
- **Media-only messages**:
  - DM: "📷 Media"
  - Group: "📷 {Sender name}: Media"

image

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [x] 🧪 Tests

## Checklist

- [x] Run `just precommit` to ensure that formatting and linting are correct
- [ ] Run `just check-flutter-coverage` to ensure that flutter coverage rules are passing
- [ ] Updated the `CHANGELOG.md` file with your changes (if they affect the user experience)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved notification formatting: sender/group names resolved per-message with DM vs group-aware titles and context-aware bodies (media/content indicators) and more consistent fallbacks.
  * Better error handling and logging when name resolution fails; uses a stable fallback label when metadata is unavailable.

* **Tests**
  * Added comprehensive tests covering message/invite filtering, sync timestamp management, display-name resolution edge cases, and lookback/buffer logic.

### CODE REVIEW

File: lib/domain/services/message_sync_service.dart [hunk 1]
```diff
@@ -106,16 +105,71 @@ class MessageSyncService {
       return;
     }
 
-    final String groupDisplayName = await getGroupDisplayName(groupId, activePubkey);
+    // Determine if it's a DM or group
+    bool isDM = false;
+    String groupDisplayName = 'Unknown';
+    try {
+      final groups = await activeGroups(pubkey: activePubkey);
```
Comment (josefinalliende) [inline lib/domain/services/message_sync_service.dart]: There is a method to get group information by group id, so I think there is no need to request all groups for rust crate to then call the isDirectMessageType.. 

What I see is that there is no method in the rust bridge to get just a single group, so I guess that's why you are fetching all? I think it would be nice to have one @jgmontoya  in the wn crate

In the meantime, I think that this fetch of all active groups is only really needed if the group is not a DM, cause the  _resolveDmDisplayName is not using the group param.

**Review by josefinalliende — COMMENTED** [review]
Left a few comments! I think the tranlsations one could be addressed in a separate issue. 

Also it would be nice to add tests for this logic

File: lib/domain/services/message_sync_service.dart (hunk 1 above)
Comment (untreu2) [reply lib/domain/services/message_sync_service.dart]: Thanks!

**Review by josefinalliende — APPROVED** [review]
✅  Thanks for adding tests 🔥 

Tested it in iPhone this cases: DMs invites, groups invites, messages with only text content, messages with media without content, and messages with media and content. All worked perfect 💯
